Scheduler with optional guests stream implementation

tonypan2000 opened the pull request:
I just implemented a naive solution for the scheduling problem without optional guests.

tonypan2000 committed:
naive solution without optional guests

remove debugging statements

tonypan2000 committed:
implement naive optional guest scheduler with tests

fold long lines

jessrenteria commented on walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:
```diff
@@ -15,9 +15,97 @@
+    // schedule for one attendee, the person initially starts free all day
+    meetingTimes.add(TimeRange.WHOLE_DAY);
+    // go through every event, mark each event the person's involved with 
+    for (Event event : events) {
```
If you were to convert everything and use RangeSet using minutes, this code (ignoring optional attendees) could look like:

```java
RangeSet<Integer> possibleTimes = TreeRangeSet.create();
possibleTimes.add(Range.closedOpen(0, 24 * 60));
possibleTimes.removeAll(
    events.stream()
        .filter(event -> !Collections.disjoint(event.getAttendees(), attendees))
        .map(FindMeetingQuery::helperMethodConvertingEventToRange)
        .collect(ImmutableRangeSet.toRangeSet()));
return possibleTimes.asRanges()
    .stream()
    .map(FindMeetingQuery::helperMethodConvertingRangeToTimeRange)
    .collect(Collectors.toList());
```

tonypan2000 committed:
switch to stream implementation without optional

add import statements

fix generic type error

fix compilation issues

add filter for duration

change to treeset

specify treerangeset types

fix casting issue

tonypan2000 committed:
stream implementation with optinal guests

fix space

add helper function

add UI support

tonypan2000 commented on walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:
```diff
@@ -15,9 +15,97 @@
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Base cases that should return default results
+    // if there is no attendee, return the whole day as available
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalGuests = request.getOptionalAttendees();
```
I think it may be because they get MeetingRequest from the client end. In the client end, the attendees are stored as an ArrayList, so they had to use as broad of a definition as possible.
